src/checkpoint.hh

```diff
@@ -1007,8 +1007,22 @@ SUPPORTED_EXTERNAL_TYPES = {"Rope", "answer_pknot_mfe", "pktype",
      stream << indent() << "#ifdef _OPENMP" << endl;
      stream << indent() << "  array_out << outer_loop_1_idx;" << endl;
      stream << indent() << "  array_out << outer_loop_2_idx;" << endl;
+
+     /*
+      * need to subtract the tile size from inner loop idx, because it is
+      * always incremented by the tile size one too many times in the
+      * parallel OpenMP loop; due to the way the parallelization works,
+      * simply subtracting the tile size once before archiving the index
+      * is the easiest workaround for this
+      */
+     stream << indent() << "#ifdef TILE_SIZE" << endl;
+     stream << indent() << "inner_loop_2_idx -= TILE_SIZE;" << endl;
```

Comment:
this looks rather ugly. Isn't there a nice way to subtract -1 (times tile_size) in the cyk.cc code?

Reply:
There probably is. I will try to clean this up.

Reply:
Fixed this with latest commit.